clang/tools/3c/README.md

```diff
@@ -53,45 +59,89 @@ of `.c` files and the options for each ([how to do this depends on the
+written. For example, this command:
 
-## More about file handling and compiler options
+```
+3c -addcr -alltypes -output-postfix=checked foo.c bar.c --
+```
 
-As an additional safeguard, `f.checked.c` or `f.checked.h` is not
-written if it is outside the _base directory_, which defaults to the
-working directory but can be overridden with the `-base-dir` option.
-This can help ensure that you don't unintentionally modify external
-libraries (though if their header files are not [annotated for Checked
-C](#annotated-headers), your ability to convert your own program to
-Checked C may be limited). However, there is a bug in the way the base
-directory check handles `..` path components in `-I` directories and
-`#include` paths (terrible, we know!), so until we can fix the bug,
-please avoid using `..` path components (e.g., use `-I` with an
-absolute path instead).
+will write the new versions to `foo.checked.c` and `bar.checked.c`. If
+one of these files is not created, it means the original file needs no
+changes. If `foo.c` includes a header file `foo.h`, then `3c` will
+automatically include it in the conversion and write the new version
+of it to `foo.checked.h` if there are changes; you don't need to
+specify header files separately on the command line, just as they
+wouldn't have their own entries in a compilation database.
 
-You can ignore the errors about a compilation database not being
-found. You can specify a single set of compiler options to use for all
-files by prefixing them with `-extra-arg-before=`, for example:
+The `-output-postfix` mode may be convenient for running tools such as
+`diff` on individual files by hand. For heavier-duty scenarios, this
+command:
 
 ```
-3c -alltypes -output-postfix=checked -extra-arg-before=-Isome/include/path foo.c bar.c
+3c -addcr -alltypes -output-dir=/path/to/new foo.c bar.c --
 ```
 
-(If you were using a compilation database, such "extra" options would
-be added to any compiler options in the database.)
+will write the new versions to `/path/to/new/foo.c` and
+`/path/to/new/bar.c`. You can then run something like `diff -ru .
+/path/to/new` to diff all the files at once (_without_ the `-N` option
+because many files in your starting directory may not have new
+versions written out).
+
+We typically recommend using the `-addcr` and `-alltypes` options, as
+shown above. Here's what they mean:
+
+- `-addcr` makes 3C add _checked region_ annotations to your code,
+  which make it easier to visualize the parts where "unsafe"
+  operations are being performed; we plan to make it the default as
+  soon as it is stable enough.
+
+- By default, 3C tries to ensure that its output always passes the
+  Checked C type checker. The `-alltypes` option enables some 3C
+  features that try to generate annotations that are closer to what
+  you ultimately want but may not pass the type checker right away
+  without manual corrections. Currently, the only such feature is
+  inference of array types.
```

Comment:
Change the last sentence to be "Currently, setting `-alltypes` is required for `3c` to attempt to infer `_Array_ptr` and `_Nt_array_ptr` types and their bounds; otherwise it will only infer `_Ptr` types. The determination that a pointer should be an array type is generally reliable, but the inferred bound may be incorrect."

Reply:
I want to emphasize that we envision `-alltypes` being the overall switch for a set of features that could in principle have individual switches too, but currently the only feature in that set is inference of array types.  (Is that your understanding too?)  So I kept the existing text but added some further explanation about array types.  OK?

Reply:
Sure that's fine. I just want you to positively say what `-alltypes` does (adds array types, which you will not get otherwise), rather than focusing on the fact that it might produce non-compilable code.